spec/download_spec.rb

```diff
@@ -11,8 +10,12 @@
 
   it "downloads a csv file with given controller and given objects" do
     objs = Product.all
-    # text = @exporter.to_csv(objs)
-    @exporter.download(@products_controller, objs)
-    # @products_controller.expects(:send_data).with(text).returns(:success)
+    text = @exporter.to_csv(objs)
+    opt = {filename: nil}
+
+    allow(@products_controller).to receive(:send_data).with(text, opt).and_return(true)
```

Comment:
You're onto something here, but you're not really testing anything quite yet.

This is a good attempt, but I see one big issue. You arbitrarily specify for the mock to return true. From what I gather, `send_data` on `ActionController::Base` does not return a boolean (although, I haven't quite figured out what it does return).

1.